README.md
# Slider Button v2.0.0
![Open Source Love](https://badges.frapsoft.com/os/v2/open-source.svg?v=103)
## Forked from [slider_button](https://github.com/anirudhsharma392/Slider-Button)
This package provides an easy implementation of a Slider Button to cancel current transaction or screen.
Highly customizable iphone alike looking widget.

<p>
    <img src="https://github.com/anirudhsharma392/Slider-Button/blob/master/screenshots/button_3.gif?raw=true"/>

</p>

### Null safety supported

## How to use
### Type 1
This is a Circle type widget.
You just need to define the border radius to swap between rectangle and circle type widget.

<p>
    <img src="https://github.com/anirudhsharma392/Slider-Button/blob/master/screenshots/button_1.gif?raw=true"/>

</p>

```dart
import 'package:forked_slider_button/forked_slider_button.dart';

```

```dart
Center(child: SliderButton(
      action: () {
        ///Do something here
        Navigator.of(context).pop();
      },
       label: Text(
          "Slide to cancel Event",
          style: TextStyle(
              color: Color(0xff4a4a4a), fontWeight: FontWeight.w500, fontSize: 17),
        ),
      icon: Text(
        "x",
        style: TextStyle(
          color: Colors.white,
          fontWeight: FontWeight.w400,
          fontSize: 44,
        ),
      ),


    ));

```
### Type 2

This is a Rectangle type widget.

<p>
    <img src="https://github.com/anirudhsharma392/Slider-Button/blob/master/screenshots/button_2.gif?raw=true"/>

</p>

```dart
import 'package:forked_slider_button/forked_slider_button.dart';

```

```dart
SliderButton(
      action: () {
        ///Do something here OnSlide
      },

      ///Put label over here
      label: Text(
        "Slide to cancel !",
        style: TextStyle(
            color: Color(0xff4a4a4a),
            fontWeight: FontWeight.w500,
            fontSize: 17),
      ),
      icon: Center(
          child: Icon(
        Icons.power_settings_new,
        color: Colors.white,
        size: 40.0,
        semanticLabel: 'Text to announce in accessibility modes',
      )),

      ///Change All the color and size from here.
      width: 230,
      radius: 10,
      buttonColor: Color(0xffd60000),
      backgroundColor: Color(0xff534bae),
      highlightedColor: Colors.white,
      baseColor: Colors.red,
    );
```

## Custom Usage
There are several options that allow for more control:

|  Properties  |   Default   |   Description   |
|--------------|-----------------|--------------|
| `action` | null | (required) Define an action after slidding a button |
| `child` | null | For more customizable button add your own widget |
| `vibrationFlag` | false | controls vibration on successful dismissed |
| `height` | null ?? 70 | Gives a height to a widget |
| `width` | null ?? 250 | Gives a width to a widget |
| `backgroundColor` | Color(0xffececec) | Gives a background color to a widget |
| `baseColor` | Color(0xff4a4a4a) | Gives a shimmer base color to a widget |
| `highlightedColor` | Colors.white | Gives a shimmer highlighted color  to a widget |
| `buttonColor` | Colors.black | Gives a color to a slidder button |
| `label` | null | A text widget which assigns a label. |
| `alignLabel` | Alignment(0.4, 0) | Aligns text label |
| `boxShadow` | null | Gives a shadow to a slidder button |
| `icon` | null | A widget to provide an icon to a button |
| `shimmer` | false | enables/disables shimmer effect on the label |
| `dismissible` | true | Make it false if you want maintain the widget in the tree |
| `buttonSize` | null ?? 60 | Gives size to a button |
| `dismissThresholds` | 1.0 | The offset threshold when it should be considered dismissed |

<br>
<br>


# 👍 Contribution
1. Fork it
2. Create your feature branch (git checkout -b my-new-feature)
3. Commit your changes (git commit -m 'Add some feature')
4. Push to the branch (git push origin my-new-feature)
5. Create new Pull Request


CHANGELOG.md
* TODO: 
add a gradient to the button.
add a spring effect on tapping a handler of a button

## [2.0.2] : 05/04/23.
* fixed wide buttons issue
* changed label type to widget

## [0.6.0] : 06/09/20.

* Fixed No implementation found for method channel vibration

## [0.5.0] : 09/03/20.

* added option to add your own button as a child widget for more customization 
* vibration is now stable

## [0.4.0] : 25/11/19.

* added buttonSize and vibration capabilities to button, Make vibrationFlag false if you want disable vibration


## [0.3.0] : 15/10/19.

* added dismissible, Make it false if you want maintain the widget in the tree


## [0.2.1] : 15/10/19.







example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:forked_slider_button/forked_slider_button.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(home: MainApp());
  }
}

class MainApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.center,
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            SliderButton(
              action: () {
                ///Do something here OnSlide
                print("working");
              },

              ///Put label over here
              label: Text(
                "Slide to cancel !",
                style: TextStyle(
                    color: Color(0xff4a4a4a),
                    fontWeight: FontWeight.w500,
                    fontSize: 17),
              ),
              icon: Center(
                  child: Icon(
                Icons.power_settings_new,
                color: Colors.white,
                size: 40.0,
                semanticLabel: 'Text to announce in accessibility modes',
              )),

              //Put BoxShadow here
              boxShadow: BoxShadow(
                color: Colors.black,
                blurRadius: 4,
              ),

              //Adjust effects such as shimmer and flag vibration here
              shimmer: true,
              vibrationFlag: true,

              ///Change All the color and size from here.
              width: 230,
              radius: 10,
              buttonColor: Color(0xffd60000),
              backgroundColor: Color(0xff534bae),
              highlightedColor: Colors.white,
              baseColor: Colors.red,
            ),
            Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text("Example of a wide button"),
            ),
            SliderButton(
              action: () {
                ///Do something here OnSlide
                print("working");
              },

              ///Put label over here
              label: Text(
                "Slide to cancel !",
                style: TextStyle(
                    color: Color(0xff4a4a4a),
                    fontWeight: FontWeight.w500,
                    fontSize: 17),
              ),
              buttonWidth: 84,
              height: 44,
              buttonSize: 44,
              child: Container(
                width: 80,
                margin: EdgeInsets.all(2),
                decoration: BoxDecoration(
                  color: Colors.red,
                  borderRadius: BorderRadius.circular(10),
                ),
                child: Icon(
                  Icons.power_settings_new,
                  color: Colors.white,
                  size: 40.0,
                  semanticLabel: 'Text to announce in accessibility modes',
                ),
              ),

              //Put BoxShadow here
              boxShadow: BoxShadow(
                color: Colors.black,
                blurRadius: 4,
              ),

              //Adjust effects such as shimmer and flag vibration here
              shimmer: true,
              vibrationFlag: true,

              ///Change All the color and size from here.
              width: 230,
              radius: 10,
              buttonColor: Color(0xffd60000),
              backgroundColor: Color(0xff534bae),
              highlightedColor: Colors.white,
              baseColor: Colors.red,
            ),
          ],
        ),
      ),
    );
  }
}
